napari/utils/perf/_timers.py

```diff
@@ -83,16 +74,34 @@ def add_event(self, event: PerfEvent):
             else:
                 self.timers[name] = Stat(duration_ms)
 
-    def add_instant_event(self, name: str, **kwargs):
+    def add_instant_event(self, name: str, **kwargs) -> None:
         """Add one instant event.
 
         Parameters
         ----------
         event : PerfEvent
             Add this event.
+        kwargs
+            Arguments to display in the Args section of the Tracing GUI.
         """
         self.add_event(InstantEvent(name, perf_counter_ns(), **kwargs))
 
+    def add_counter_event(self, name: str, **kwargs: Dict[str, float]) -> None:
+        """Add one counter event.
+
+        Parameters
+        ----------
+        name : str
+            The name of this event like "draw".
+        kwargs : Dict[str, float]
+            The individual counters for this event.
+
+        Notes
+        -----
+        For example add_counter_event("draw", triangles=5, squares=10).
+        """
+        self.add_event(CounterEvent(name, perf_counter_ns(), **kwargs))
```

Comment:
I think how it is is better because then the line of code says CounterEvent. Your way "counter event" is no present in the code at all, the read has to know that phase "C" is a counter event.